x-pack/plugins/apm/server/routes/settings/anomaly_detection/route.ts

```diff
@@ -29,22 +29,21 @@ const anomalyDetectionJobsRoute = createApmServerRoute({
   },
   handler: async (resources) => {
     const setup = await setupRequest(resources);
-    const { context, logger } = resources;
+    const { context } = resources;
 
     if (!isActivePlatinumLicense(context.licensing.license)) {
       throw Boom.forbidden(ML_ERRORS.INVALID_LICENSE);
     }
 
-    const [jobs, legacyJobs] = await withApmSpan('get_available_ml_jobs', () =>
-      Promise.all([
-        getAnomalyDetectionJobs(setup, logger),
-        hasLegacyJobs(setup),
-      ])
-    );
+    if (!setup.ml) {
+      throw Boom.forbidden(ML_ERRORS.ML_NOT_AVAILABLE);
+    }
+
+    const jobs = await getMlJobsWithAPMGroup(setup.ml?.anomalyDetectors);
```

Comment:
There is a helper `getAnomalyDetectionJobs`. All it does is calling through to `getMlJobsWithAPMGroup` if `setup.ml` is defined, else it'll throw like here.

Question: should we use the helper here? If not, perhaps the helper is not so useful and we can delete it.